src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java

```diff
@@ -98,30 +98,30 @@ public void connect(MessageContext messageContext) {
      * Method to validate the MT940 format.
      *
      * @param messageContext  Message context
-     * @param lines           Payload lines
-     * @return                Error model
+     * @param lines           Payload lines containing the Mt940 fields
+     * @return        Validation Result with error details if there is an error, else empty Validation Result
      */
-    private ErrorModel validateMT940(MessageContext messageContext, String[] lines) {
+    private ValidationResult validateMT940(MessageContext messageContext, String[] lines) {
         Map<String, Object> extractFields = extractFields(lines);
```

Comment:
Is it must to use a Map<String, **Object**> here? If possible, use a custom model class instead of the map to ensure type safety.

Reply:
This will be considered in another project by Izzath